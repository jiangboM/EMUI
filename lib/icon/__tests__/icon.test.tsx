// import jest from '@types/jest'
import * as renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import React from "react";
import Icon from '../icon'

describe('icon', ()=> {
   it('is SVG', () => {
       const json = renderer.create(<Icon name='wechat' />).toJSON();
       expect(json).toMatchSnapshot()
   });
    it( 'asset onclick', () => {
        const fn = jest.fn();
        const c = mount(<Icon name='wechat' onClick={fn} />);
        c.find('svg').simulate('click');
        expect(fn).toBeCalled()
    })
});