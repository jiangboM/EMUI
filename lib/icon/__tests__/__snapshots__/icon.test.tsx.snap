// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon is SVG 1`] = `
<svg
  className="icon-wrap"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
